googlemock/include/gmock/gmock-actions.h

```diff
@@ -617,9 +617,8 @@ class ReturnNullAction {
 class ReturnVoidAction {
  public:
   // Allows Return() to be used in any void-returning function.
-  template <typename Result, typename ArgumentTuple>
+  template <typename Result, typename ArgumentTuple, typename = std::enable_if<std::is_same<Result, void>::value>>
```

Comment:
I cannot make a suggestion here as it's multiple-liner, sorry.
This has the following implications:
1) Compiles longer (SFINAE is a performance killer)
2) Instead of giving a clear compile time error message, it will provide "couldn't find a function"
3) It is a bit harder to maintain than a plain static assert
4) There is no reason to SFINAE since you've got no additional overload of the Perform() function in this structure.
Please use a static assertion. In addition, std::is_same could be turned into std::is_void.